applications/luci-app-shadowsocks-libev/htdocs/luci-static/resources/view/shadowsocks-libev/servers.js

```diff
@@ -1,21 +1,61 @@
 'use strict';
 'require form';
+'require uci';
+'require ui';
 'require shadowsocks-libev as ss';
 
-function startsWith(str, search) {
-	return str.substring(0, search.length) === search;
-}
+var conf = 'shadowsocks-libev';
 
 return L.view.extend({
 	render: function() {
 		var m, s, o;
 
-		m = new form.Map('shadowsocks-libev', _('Remote Servers'),
+		m = new form.Map(conf, _('Remote Servers'),
 			_('Definition of remote shadowsocks servers.  \
 				Disable any of them will also disable instances referring to it.'));
 
 		s = m.section(form.GridSection, 'server');
 		s.addremove = true;
+		s.handleLinkImport = function() {
+			var textarea = new ui.Textarea();
+			ui.showModal(_('Import Links'), [
+				textarea.render(),
+				E('div', { class: 'right' }, [
+					E('button', {
+						class: 'btn',
+						click: ui.hideModal
+					}, [ _('Cancel') ]),
+					' ',
+					E('button', {
+						class: 'btn cbi-button-action',
+						click: ui.createHandlerFn(this, function() {
+							textarea.getValue().split('\n').forEach(function(s) {
+								var config = ss.parse_uri(s);
+								if (config) {
+									var tag = config[1];
+									if (tag && !tag.match(/^[a-zA-Z0-9_]+$/)) tag = null;
+									var sid = uci.add(conf, 'server', tag);
+									config = config[0];
+									Object.keys(config).forEach(function(k) {
+										uci.set(conf, sid, k, config[k]);
+									});
+								}
+							});
+							return uci.save().finally(ui.hideModal);
```

Comment:
Can we reload the view after the addition succeeded?

Reply:
OK.

Reply:
Have you tried it?  It does not work here.  Only "name" and "method" were set, other options like server address, port, etc. were missing.

Reply:
Sorry for my mistake.